packages/x-charts/src/internals/plugins/corePlugins/useChartSeries/useChartSeries.ts

```diff
@@ -49,8 +49,8 @@ useChartSeries.params = {
 const EMPTY_ARRAY: any[] = [];
 
 useChartSeries.getDefaultizedParams = ({ params }) => ({
-  series: EMPTY_ARRAY,
   ...params,
+  series: params.series?.length ? [...params.series] : EMPTY_ARRAY,
```

Comment:
Why do we need this change? If `params.series` is an `Array`, it should be compatible with a `readonly Array` without any change, shouldn't it?

Reply:
Not really. Our internal state series is defined as `mutabled`, a `readonly` array can't be assigned to `mutable`. The other way around works though.

Reply:
Should we fix it from the inside out, then? Or maybe as a follow-up? 

It feels weird to me that we're creating a new array just to appease TypeScript. 🤔

Reply:
Technically we are promising the user the array they don't pass in won't be mutated. Hence the `...` we can def solve it in other ways, as generally that array is not mutated though